src/rust/inetstack/protocols/layer4/tcp/established/sender.rs

```diff
@@ -221,16 +221,22 @@ impl Sender {
         debug_assert!(cb.sender.fin_seq_no.is_none());
 
         // TODO: We need to fix this the correct way: limit our send buffer size to the amount we're willing to buffer.
-        if cb.sender.unsent_queue.len() > UNSENT_QUEUE_CUTOFF {
+        if cb.sender.unsent_queue.len() > UNSENT_QUEUE_CUTOFF - 1 {
             return Err(Fail::new(libc::EBUSY, "too many packets to send"));
         }
 
+        trace!("push(): unsent_queue.len() = {:?}", cb.sender.unsent_queue.len());
+
         // Place the buffer in the unsent queue.
         cb.sender.unsent_next_seq_no = cb.sender.unsent_next_seq_no + (buf.len() as u32).into();
         if cb.sender.send_window.get() > 0 {
             Self::send_segment(cb, layer3_endpoint, runtime.get_now(), &mut buf);
         }
         if buf.len() > 0 {
+            if cb.sender.unacked_queue.len() > 0 {
+                trace!("push(): unacked_queue.len() = {:?}", cb.sender.unacked_queue.len());
```

Comment:
It would be good to differentiate between this and the identical trace msg above. They might optionally show up as consecutive traces in the logs.

Reply:
I will change it. For future reference, our traces do print the file and line number if we need to differentiate. But I agree that it could be confusing